### PR TITLE
Add basic auth support

### PR DESCRIPTION
**Pull request by justinsteven**
I left room for alternate auth schemes to be added ala curl

`--basic` could be added, and be the default. `--ntlm` and `--digest` could be added to switch methods.

### CODE REVIEW

**Comment by OJ** [comment]
Thanks for the PR! I shall have a look at it shortly.

**Comment by OJ** [comment]
I haven't forgotten about this by the way @justinsteven :)

File: main.go [hunk 1]
```diff
@@ -242,6 +250,14 @@ func ParseCmdLine() *State {
 			}
 		}
 
+		// prompt for password if needed
+		if valid && s.Username != "" && s.Password == "" {
+			print("Auth Password: ")
+			reader := bufio.NewReader(os.Stdin)
+			s.Password, _ = reader.ReadString('\n')
+			s.Password = s.Password[:len(s.Password)-1]
```
Comment (OJ) [inline main.go]: Did you test this out on Windows? I'm tempted to use [this solution](http://stackoverflow.com/a/32768479/611) instead. Thoughts?

Comment (justinsteven) [inline main.go]: No, I didn't. Good call :)

**Comment by OJ** [comment]
This hasn't been forgotten @justinsteven ;) I'll be working on this by the end of the week and including a couple of other PRs too. Cheers!

**Pull request merged**